packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -484,8 +484,10 @@ class ScaffoldState extends State<Scaffold> {
       List<Widget> bottomSheets = <Widget>[];
       if (_dismissedBottomSheets != null && _dismissedBottomSheets.isNotEmpty)
         bottomSheets.addAll(_dismissedBottomSheets);
-      if (_currentBottomSheet != null)
+      if (_currentBottomSheet != null) {
+        _currentBottomSheet.setState(() { });
```

Comment:
I think the right thing to do here is to have `_PersistentBottomSheet` have a `clone()` method that returns a new identical widget, and put that one into the bottomSheets list instead of _currentBottomSheet._widget.